### PR TITLE
Use ShadowCopy to load tests

### LINKED ISSUE #170: Test assembly locked
Would maybe be resolved by #43 (reload on change), but currently the GUI locks the Test Assembly and so cannot be overwritten during the build without closing the application.

### PR DESCRIPTION
**Pull request by CharliePoole**
Fixes #170

### CODE REVIEW

File: src/nunit-gui/Model/TestModel.cs [hunk 1]
```diff
@@ -240,8 +240,7 @@ public void UnloadTests()
             _files = null;
             _resultIndex.Clear();
 
-            if (TestUnloaded != null)
-                TestUnloaded(new TestEventArgs(TestAction.TestUnloaded));
```
Comment (mikkelbu) [inline src/nunit-gui/Model/TestModel.cs]: Why the removal of the guard? (I guess that right now we can be sure that there are handlers associated with the event, but that might not be case in the future?)

File: src/nunit-gui/Views/ProgressBarView.Designer.cs [hunk 2]
```diff
@@ -55,6 +55,5 @@ private void InitializeComponent()
         #endregion
 
         private NUnit.UiKit.Controls.NUnitProgressBar testProgressBar;
-        private System.Windows.Forms.Label testStatusesLabel;
```
Comment (mikkelbu) [inline src/nunit-gui/Views/ProgressBarView.Designer.cs]: Seems unrelated, but I'm fine with "cleaning up" :)

**Comment by mikkelbu** [comment]
Looks correct. I only have some minor questions, but please merge :)

File: src/nunit-gui/Views/ProgressBarView.Designer.cs (hunk 2 above)
Comment (CharliePoole) [inline src/nunit-gui/Views/ProgressBarView.Designer.cs]: Warning gives an error on travis, so it has to be cleaned.

File: src/nunit-gui/Model/TestModel.cs (hunk 1 above)
Comment (CharliePoole) [inline src/nunit-gui/Model/TestModel.cs]: Actually, I intended to use ? operator. Good catch. I'll check other events as well.

File: src/nunit-gui/Model/TestModel.cs [hunk 3]
```diff
@@ -350,39 +348,33 @@ public void OnTestEvent(string report)
...
-                        TestFinished(new TestResultEventArgs(TestAction.TestFinished, result));
+                    TestFinished?.Invoke(new TestResultEventArgs(TestAction.TestFinished, result));
                     break;
 
                 case "test-suite":
                     result = new ResultNode(xmlNode);
                     _resultIndex[result.Id] = result;
-                    if (SuiteFinished != null)
-                        SuiteFinished(new TestResultEventArgs(TestAction.TestFinished, result));
+                    SuiteFinished?.Invoke(new TestResultEventArgs(TestAction.TestFinished, result));
```
Comment (mikkelbu) [inline src/nunit-gui/Model/TestModel.cs]: All the translations to conditional member access looks correct (and also we are now threadsafe ;) ). The only problem I discovered was that we are using the wrong `TestAction`. It should be `TestAction.SuiteFinished` not `TestAction.TestFinished`.

Comment (CharliePoole) [inline src/nunit-gui/Model/TestModel.cs]: Dang!

All the translations are correct, 'cause I let VS do it this time. 😄

**Pull request merged**